### PR TITLE
#317 : Adding Categories of Microservices for Beginners

### PR DESCRIPTION
**Pull request by JonasAls89**
Correcting error on ref line inn Systems -> Beginner.rst file when making seealso section on this PR.

### CODE REVIEW

File: training/050_microservices/020_Beginner.rst [hunk 1]
```diff
@@ -80,24 +80,47 @@ Pipe source/sink/http
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   Interne
+Extending on the topic of running microservices in Sesam, the following categories of microservices will be elaborated on in this section:
+  
+  - Internal microservices
+  - External microservices
 
-   http-transform
+Both internal microservices and external microservices will figurate in a Sesam node as systems. However, microservices that connect to a Sesam node via the `Service API <https://docs.sesam.io/api.html>`_ can also be seen on the `Sesam Community on GitHub <https://github.com/sesam-community>`_. These kinds of microservices will not be elaborated on in this section, albeit you should be aware that these microservices exist and that they are also a viable option when extending upon default functionality.
```
Comment (ga-hegsvold) [inline training/050_microservices/020_Beginner.rst:93]: Is "figurate" a good term to use here? https://www.dictionary.com/browse/figurate

Can we use "be defined" instead?

File: training/050_microservices/020_Beginner.rst [hunk 2]
```diff
@@ -80,24 +80,47 @@ Pipe source/sink/http
...
+Internal microservices
+######################
+
+These kinds of microservices are the ones used with regards to Sesam dataflows. Therefore, these are also the most commonly used. Internal microservices can be used at specific points in time with regards to a dataflow. As such, internal microservices can be used in the beginning, in the middle or at the end of a dataflow, which will respectively, in a Sesam node, turn out as a source system, http-transform system or sink system. Many of these internal microservices can be found on the `Sesam Community on GitHub <https://github.com/sesam-community>`_.
+
+
+External microservices
+######################
+
+Even though external microservices figurate in Sesam as systems, they do not read or write data into or out of Sesam with regards to a defined dataflow. Rather, they implement their functionality in such a way that does not affect dataflows.
```
Comment (ga-hegsvold) [inline training/050_microservices/020_Beginner.rst:105]: "figurate"? - see comment above.

**Review by ga-hegsvold — CHANGES_REQUESTED** [review]
Nice! I just have a comment about the term "figurate".

File: training/050_microservices/020_Beginner.rst (hunk 2 above)
Comment (JonasAls89) [reply training/050_microservices/020_Beginner.rst:105]: Oppdatert.

Comment (ga-hegsvold) [reply training/050_microservices/020_Beginner.rst:105]: Hm.. Still says "figurate" here on my end... Are all changes pushed?

**Review by ga-hegsvold — CHANGES_REQUESTED** [review]
One "figurate" remaining on my end... See below.

File: training/050_microservices/020_Beginner.rst (hunk 2 above)
Comment (JonasAls89) [reply training/050_microservices/020_Beginner.rst:105]: missed that one. two sec

File: training/050_microservices/020_Beginner.rst [hunk 3]
```diff
@@ -80,24 +80,47 @@ Pipe source/sink/http
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   Interne
+Extending on the topic of running microservices in Sesam, the following categories of microservices will be elaborated on in this section:
+  
+  - Internal microservices
+  - External microservices
 
-   http-transform
+Both internal microservices and external microservices will be defined in a Sesam node as systems. However, microservices that connect to a Sesam node via the `Service API <https://docs.sesam.io/api.html>`_ can also be seen on the `Sesam Community on GitHub <https://github.com/sesam-community>`_. These kinds of microservices will not be elaborated on in this section, albeit you should be aware that these microservices exist and that they are also a viable option when extending upon default functionality.
```
Comment (ga-hegsvold) [inline training/050_microservices/020_Beginner.rst:93]: This is not a big deal but I think it will improve the flow of reading a bit: At the beginning of this sentence, the repetition of "microservices" after both "internal" and "external" is not needed.
"Both internal and external microservices..." flows a bit better, I think.

**Review by ga-hegsvold — APPROVED** [review]
See my comment below. It's a nit-pick. Adjust if you agree. I'll approve regardless.

File: training/050_microservices/020_Beginner.rst (hunk 3 above)
Comment (JonasAls89) [reply training/050_microservices/020_Beginner.rst:93]: onn its way :yoda:

File: training/050_microservices/020_Beginner.rst [hunk 4]
```diff
@@ -80,24 +80,47 @@ Pipe source/sink/http
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   Interne
+Extending on the topic of running microservices in Sesam, the following categories of microservices will be elaborated on in this section:
+  
+  - Internal microservices
+  - External microservices
 
-   http-transform
+Both internal and external microservices will be defined in a Sesam node as systems. However, microservices that connect to a Sesam node via the `Service API <https://docs.sesam.io/api.html>`_ can also be seen on the `Sesam Community on GitHub <https://github.com/sesam-community>`_. These kinds of microservices will not be elaborated on in this section, albeit you should be aware that these microservices exist and that they are also a viable option when extending upon default functionality.
```
Comment (gabriellsesam) [inline training/050_microservices/020_Beginner.rst:93]: `Both internal and external microservices will be defined in a Sesam node as systems.`
external microservices do not need to be defined in a Sesam node as systems, they can be run outside of Sesam also.

File: training/050_microservices/020_Beginner.rst [hunk 5]
```diff
@@ -80,24 +80,47 @@ Pipe source/sink/http
...
+Internal microservices
+######################
+
+These kinds of microservices are the ones used with regards to Sesam dataflows. Therefore, these are also the most commonly used. Internal microservices can be used at specific points in time with regards to a dataflow. As such, internal microservices can be used in the beginning, in the middle or at the end of a dataflow, which will respectively, in a Sesam node, turn out as a source system, http-transform system or sink system. Many of these internal microservices can be found on the `Sesam Community on GitHub <https://github.com/sesam-community>`_.
+
+
+External microservices
+######################
+
+Even though external microservices are defined in Sesam as systems, they do not read or write data into or out of Sesam with regards to a defined dataflow. Rather, they implement their functionality in such a way that does not affect dataflows.
```
Comment (gabriellsesam) [inline training/050_microservices/020_Beginner.rst:105]:
```
Even though external microservices are defined in Sesam as systems, they do not...
```
I would rephrase to (building on my last comment)
```
External microservices can be hosted inside your Sesam node as a system or outside of your Sesam node in your preferred docker-compatible solution like for example Kubernetes.
They do not ...
```

**Review by gabriellsesam — CHANGES_REQUESTED** [review]
I want this section to make it clear that External microservices can run outside of a Sesam node, otherwise very well written.

File: training/050_microservices/020_Beginner.rst (hunk 4 above)
Comment (JonasAls89) [reply training/050_microservices/020_Beginner.rst:93]: On its way! 👍

**Review by gabriellsesam — APPROVED** [review]

**Review by mofsesam — APPROVED** [review]
I have no problem with the changes, but in more general I wonder in the introduction of concepts of internal and external microservices helps a beginner.
